pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -84,8 +86,17 @@ func (s *preFilterState) Clone() framework.StateData {
 // EventsToRegister returns the possible events that may make a Pod
 // failed by this plugin schedulable.
 func (pl *NodeAffinity) EventsToRegister(_ context.Context) ([]framework.ClusterEventWithHint, error) {
+	// Ideally, it's supposed to register only Add|UpdateNodeLabel, because the only label update could change the node affinity result.
+	// But, we may miss some Node's label update event due to the problematic preCheck.
+	// See: https://github.com/kubernetes/kubernetes/issues/109437
+	nodeActionType := framework.Add | framework.Update
```

Comment:
Can it be like the others?
```suggestion
	nodeActionType := framework.Add | framework.UpdateNodeLabel | framework.UpdateNodeTaint
```

Reply:
Good point. I think we can do that. 
Although your change would contain the risk of some Pods stucking in the queue for 5 mins very rarely, we can/should accept it because we've already accepted the risk in the in-tree plugins that already only have `Node/UpdateTaint`, not `Node/Update`.

Reply:
Assuming that the preChecks don't check anything other than labels and taints, I think there is no risk.

Reply:
No, it checks the port and resources too. So, it still has risk, unless we set `framework.Update` here.

Reply:
> don't we need UpdateNodeLabel here too?

Responding other your comments here to merge the discussion.

So, how many node events we cover here would equal how much we want to go to the safer way. 
And, if we need a complete 100% safe, then we need to have `framework.Update` (because we don't have NodePortUpdate event).

The reason I added only `updateNodeTaint` is, I thought we understood this risk, and made a decision to only add `updateNodeTaint` in this case, instead of a general `framework.Update` at https://github.com/kubernetes/kubernetes/pull/122292.
If we want to go more conservative, we can change this direction though. What do you think

Reply:
Right, thanks for the refresher.

I agree that the safest would be to do `framework.Update` and we can look for a compromise.
Note that we are more likely to miss events (because of prechecks) when the node is starting up. When starting up, the common things to happen are for taints to be removed and labels to be added. So I think we to cover those to be on the safe side.

> The reason I added only updateNodeTaint is, I thought we understood this risk, and made a decision to only add updateNodeTaint in this case, instead of a general framework.Update at https://github.com/kubernetes/kubernetes/pull/122292.

It looks like the previous decision was to keep `framework.Update` in node affinity, specifically?
I think we decided to just keep updateNodeTaint in some other plugins just because we had no bug reports for those plugins.

Reply:
Okay, then let's do label+taint. I'll update all the plugins.